packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -2,17 +2,153 @@ import 'clay-button';
 import 'clay-checkbox';
 import 'clay-icon';
 import 'clay-radio';
-import Dropdown from 'metal-dropdown';
-import Soy from 'metal-soy';
+import {Align} from 'metal-position';
+import {core, object} from 'metal';
 import {Config} from 'metal-state';
-
+import Component from 'metal-component';
+import dom from 'metal-dom';
+import {EventHandler} from 'metal-events';
+import Soy from 'metal-soy';
 import itemsValidator from './items_validator';
 import templates from './ClayDropdownBase.soy.js';
 
 /**
  * Implementation of the base for Metal Clay Dropdown.
  */
-class ClayDropdownBase extends Dropdown {
+class ClayDropdownBase extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		super.attached();
+		this.eventHandler_.add(
+			dom.on(document, 'click', this.handleDocClick_.bind(this))
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.eventHandler_ = new EventHandler();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	detached() {
+		super.attached();
```

Comment:
Lifecycle methods are just defined in Component, but there's nothing inside. I guess we can get rid of this super call.

https://github.com/metal/metal.js/blob/master/packages/metal-component/src/Component.js#L283